tests/extendrtests/tests/testthat/test-test-submodules.R

```diff
@@ -21,3 +21,25 @@ test_that("Classes defined in submodules also works", {
     isTRUE(identical(x$me_owned(), x))
   )
 })
+
+test_that("Return the right externalptr", {
+  x <- MySubmoduleClass$new()
+  y <- MySubmoduleClass$new()
+  x$set_a(10)
+  y$set_a(42)
+  expect_false(isTRUE(identical(x, y)))
+
+  max_class <- x$max_ref(y)
+  expect_identical(max_class, y)
+  expect_false(isTRUE(identical(x, max_class)))
+
+  # check if you can have regular arguments as well
+  x <- MySubmoduleClass$new()
+  x$set_a(22)
+  x$max_ref_offset(y, 30)
```

Comment:
Why don't you use the return value here?

Reply:
I'm using it now. Thanks!